docs/dev-client-gameplay.md

```diff
@@ -169,41 +341,70 @@ export const VideoMenu = defineComponent({
-
+### String data
 
-<a id="orgdf10967"></a>
+We sometimes want to be able to store string data in components. Since `bitECS`
+does not allow strings in components, we provide a helper mechanism to store
+numeric string ID&rsquo;s instead.
 
-## String data
+Let's think of a `SceneLoader` component with a `src` property, which we
+wish was a string, as an example.
 
-We sometimes want to be able to store string data in components. Since `bitECS` does not allow strings in components, we store numeric string ID&rsquo;s instead.
-
-For example, consider a `SceneLoader` component with a `src` property, which we wish was a string.
+Define a component with `src` property as `ui32` type. And mark the `src`
+property as string data with the symbol `$isStringType`, defined in the Hubs
+Client core `src/bit-components.js` file.
 
 ```typescript
 export const SceneLoader = defineComponent({ src: Types.ui32 });
 SceneLoader.src[$isStringType] = true;
 ```
 
-The symbol `$isStringType`, defined in `bit-components.js`, indicates that this property is a string handle. Code that handles component state anonymously (e.g. `createDefaultInflator`) use this to correctly handle the property values.
+The symbol indicates that this property is a string handle. Code that handles
+component state anonymously (e.g. [our `createDefaultInflator()` that will be
+explained later](#default-inflators)) use this to correctly handle the property values.
+
+Strings are converted to numeric `StringID` s by the `APP.getSid` function.
+`StringID` s can be converted back to strings by the `APP.getString` function.
+`APP` is exposed to global scope from Hubs Client core `src/app.ts` and it
+can be accessed anywhere.
 
-Strings are converted to numeric `StringID` s by the `getSid` function. `StringID` s can be converted back to strings by the `getString` function.
+Then, code that initializes `SceneLoader` component will be like this
 
 ```typescript
-const src = APP.getString(SceneLoader.src[loaderEid]);
-console.log(`Loading scene from this url: ${src}`);
+export type SceneLoaderParams = {
+  url: string;
+};
+
+export function inflateSceneLoader(
+  world: HubsWorld,
+  eid: number,
+  params: SceneLoaderParams
+) {
+  addComponent(world, SceneLoader, eid);
+  SceneLoader.src[eid] = APP.getSid(params.url);
+}
 ```
 
+and code that accesses the property will be like this.
+
+```typescript
+const sceneLoaderQuery = defineQuery([SceneLoader]);
+const sceneLoaderEnterQuery = enterQuery(sceneLoaderQuery);
+export function sceneLoaderSystem(world: HubsWorld) {
+  sceneLoaderEnterQuery(world).forEach(eid => {
+    const src = APP.getString(SceneLoader.src[eid]);
+    console.log(`Loading scene from this url: ${src}`);
+    ...
+  });
+}
+```
 
-<a id="org70bd5dd"></a>
 
-## Flags
+### Flags

```

Comment:
“Booleans/Flags” similar reasoning to the Coroutine section header suggestion above.

Reply:
Updated, thanks.